static/app/views/preprod/buildDetails/main/insights/insightInfoModal.tsx

```diff
@@ -0,0 +1,129 @@
+import {Fragment, type ReactNode} from 'react';
+import styled from '@emotion/styled';
+import {AnimatePresence, motion} from 'framer-motion';
+
+import {Button} from 'sentry/components/core/button';
+import {Flex} from 'sentry/components/core/layout';
+import {Heading} from 'sentry/components/core/text/heading';
+import {IconClose} from 'sentry/icons';
+import {t} from 'sentry/locale';
+
+interface InsightInfoModalProps {
+  children: ReactNode;
+  isOpen: boolean;
+  onClose: () => void;
+  title: string;
+}
+
+export function InsightInfoModal({
```

Comment:
Chatted with @JonasBa here, looks like `static/app/actionCreators/modal.tsx` is a good pattern of existing modal usage we could leverage here. Would recommend looking into that and applying the same pattern here if possible (I haven't looked into yet)